assets/js/modules/analytics/datastore/report.js

```diff
@@ -92,7 +95,15 @@ const baseResolvers = {
 			return;
 		}
 
-		yield fetchGetReportStore.actions.fetchGetReport( options );
+		const { error } = yield fetchGetReportStore.actions.fetchGetReport( options );
+
+		if (
+			error?.code === 400 &&
+			error?.message.startsWith( 'Restricted metric' ) &&
+			normalizeReportOptions( options ).metrics.some( ( { expression } ) => /^ga:adsense/.test( expression ) )
```

Comment:
There is something missing here:
> If `options.metrics` includes any `ga:adsense*` metrics, the resolver should inspect the results of the `fetchGetReport` call

The reason for inspecting `options.metrics` is that such requests should "either way" result in a call to `setAdsenseLinked`. The way you currently have it here, we could as well look at `error.message` to see if it includes any `ga:adsense*`, which would be simpler. However, that's not enough. The clause should be:
* If `options.metrics` includes any `ga:adsense*`
    * If there is a 400 Restricted metric error, `setAdsenseLinked( false )`.
    * Otherwise, `setAdsenseLinked( true )`.

Reply:
Oops, yeah I missed that part. No problem 👍